File: 167. Two Sum II - Input Array Is Sorted/167. Two Sum II - Input Array Is Sorted.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int left = 0;
        int right = numbers.length - 1;

        while (left < right) {
            int summ = numbers[left] + numbers[right];
            if (summ == target) {
                return new int[] {left + 1, right + 1};
            } else if (summ > target) {
                right--;
            } else {
                left++;
            }
        }

        throw new Error("No solution");
    }
}

File: 167. Two Sum II - Input Array Is Sorted/167. Two Sum II - Input Array Is Sorted.js
/**
 * @param {number[]} numbers
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (numbers, target) {
  let left = 0;
  let right = numbers.length - 1;

  while (left < right) {
    const summ = numbers[left] + numbers[right];
    if (summ === target) {
      return [left + 1, right + 1];
    } else if (summ > target) {
      right--;
    } else {
      left++;
    }
  }
};
